File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/IntakeServoHandler.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.hardware.Gamepad;
import com.qualcomm.robotcore.hardware.HardwareMap;

public class IntakeServoHandler {
    public final ServoWrapper servo;
    public static final double HOOKED = 0.25;
    public static final double RELEASED = 0.5;

    public IntakeServoHandler(HardwareMap map) {
        servo = ServoWrapper.get(map, "intakeServo");
    }

    public void goToPos(double pos) {
        if (pos < HOOKED) {
            servo.setPos(HOOKED);
        } else if (pos > RELEASED) {
            servo.setPos(RELEASED);
        } else {
            servo.setPos(pos);
        }
        servo.update();
    }

    public void hook() {
        goToPos(HOOKED);
    }

    public void release() {
        goToPos(RELEASED);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/IntakeServoHandler.kt
package org.firstinspires.ftc.teamcode

import com.qualcomm.robotcore.hardware.HardwareMap

class IntakeServoHandler(map: HardwareMap) {
    companion object {
        const val HOOKED = 0.25
        const val RELEASED = 0.5
    }

    @JvmField
    val servo = ServoWrapper[map, "intakeServo"]

    fun goToPos(pos: Double) {
        servo.pos = when {
            pos < HOOKED -> HOOKED
            pos > RELEASED -> RELEASED
            else -> pos
        }
        servo.update()
    }

    fun hook() {
        goToPos(HOOKED)
    }

    fun release() {
        goToPos(RELEASED)
    }
}